todo/utils.py

```diff
@@ -19,45 +23,141 @@ def staff_check(user):
         return True
 
 
-def send_notify_mail(new_task):
-    # Send email to assignee if task is assigned to someone other than submittor.
-    # Unassigned tasks should not try to notify.
-
-    if not new_task.assigned_to == new_task.created_by:
-        current_site = Site.objects.get_current()
-        email_subject = render_to_string("todo/email/assigned_subject.txt", {"task": new_task})
-        email_body = render_to_string(
-            "todo/email/assigned_body.txt", {"task": new_task, "site": current_site}
-        )
+def user_can_read_task(task, user):
+    return task.task_list.group in user.groups.all() or user.is_staff
+
+
+def todo_get_backend(task):
+    '''returns a mail backend for some task'''
+    mail_backends = getattr(settings, "TODO_MAIL_BACKENDS", None)
+    if mail_backends is None:
+        return None
+
+    task_backend = mail_backends[task.task_list.slug]
+    if task_backend is None:
+        return None
+
+    return task_backend
+
 
-        send_mail(
-            email_subject,
-            email_body,
-            new_task.created_by.email,
-            [new_task.assigned_to.email],
-            fail_silently=False,
+def todo_get_mailer(user, task):
+    """a mailer is a (from_address, backend) pair"""
+    task_backend = todo_get_backend(task)
+    if task_backend is None:
+        return (None, mail.get_connection)
+
+    from_address = getattr(task_backend, "from_address", None)
+    if from_address is None:
+        # worst fallback ever
```

Comment:
:)

Reply:
fixed in 6afd791, no fallback feels better than a terrible one